src/soc/ibm/power9/timer.c

```diff
@@ -58,6 +58,15 @@ void init_timer(void)
 {
 	uint64_t tmp;
 
+	/*
+	 * 1. Data Cache Block set to Zero for 0..0x1000
+	 * 2. Instruction Cache Block Invalidate for 0..0x1000 - single 'isync'
+	 *    before enabling interrupts by writing to MSR is enough (not defined
+	 *    by ISA but in POWER9 Processor User's Manual, 4.6.2.2)
+	 */
+	for (tmp = 0; tmp < 0x1000; tmp += 128)
+		asm volatile("dcbz 0, %0; icbi 0, %0;" :: "r"(tmp) : "memory");
+
```

Comment:
Could you mention that this initializes interrupt vectors and explain why this happens in `init_timer()`?

Reply:
Done and rebased, however I may be able to simplify it more.

I've tried to make it work without interrupts and almost got it working. `wait` instruction waits until _exception_ occurs, without the need to actually take the _interrupt_ (on POWER exception is just a possible cause for interrupt, it doesn't always mean something bad as on x86). The only issue is that there is an exception at the start (maybe that asserted IPMI?) and I couldn't find a way to differentiate between them.

Reply:
Ok, udelay() should work now and no longer needs separate implementation for ramstage